exp/query/api.go

```diff
@@ -88,44 +107,182 @@ func (a *API) Search(ctx context.Context, req *SearchRequest) (*SearchResponse,
 		return nil, fmt.Errorf("tenant %q is not offloaded, %w", req.Tenant, ErrInvalidTenant)
 	}
 
-	// src - s3://<collection>/<tenant>/<node>/
-	// dst (local) - <data-path/<collection>/<tenant>
-	if err := a.offload.Download(ctx, req.Collection, req.Tenant, nodeName); err != nil {
+	store, localPath, err := a.loadOrDownloadLSM(ctx, req.Collection, req.Tenant)
+	if err != nil {
 		return nil, err
 	}
+	defer store.Shutdown(ctx)
 
-	localPath := path.Join(a.offload.DataPath, strings.ToLower(req.Collection), strings.ToLower(req.Tenant), defaultLSMRoot)
+	var resp SearchResponse
 
-	// TODO(kavi): Avoid creating store every time?
-	store, err := lsmkv.New(localPath, localPath, a.log, nil, cyclemanager.NewCallbackGroupNoop(), cyclemanager.NewCallbackGroupNoop())
+	limit := req.Limit
+	if limit == 0 {
+		limit = maxQueryObjectsLimit
+	}
+
+	if len(req.NearText) != 0 {
```

Comment:
NIT: More of a style question, but to avoid a big `if... else...` block, would it be possible to package the code for near text in a function and the one for get objects into another and then we end up with
```
if nearText {
 return nearText(...)
}

return getObjects(...)
```

Might not be do-able or easy but I think it would improve the readability.